### PR TITLE
ztp based cumulus discover and configuration

### PR DESCRIPTION
**Pull request by immarvin**
on investigation task #3060:
For the implementation task #3176 

based on https://gist.github.com/mattaezell/f385baa5bdccf6f63145489f3c172da2 from @mattaezell 
the script to obtain cumulus switch information and send "findme" request to xcatd, intended to be invoked by the ZTP script

### CODE REVIEW

File: xCAT/postscripts/documulusdiscovery [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+fi
+
+PUBKEY=`openssl rsa -in /etc/xcat/privkey.pem -pubout 2> /dev/null|grep -v "PUBLIC KEY"`
+PUBKEY=`echo $PUBKEY|sed -e 's/ //g'`
+export PUBKEY
+
+logger -s -t $log_label -p local4.info "Beginning switch discovery process..."
+
+XCATPORT=3001
+export XCATPORT
```
Comment (zet809) [inline xCAT/postscripts/documulusdiscovery]: Why export XCATPORT here? Why not export XCATMASTER in line 4?

**Comment by zet809** [comment]
Hi, @immarvin , the scripts look good to me, one more comment:
After switch node is matched, the switch information included in 'findme' request will be updated to the objective node in nodediscover.pm. This module need some modification in order to update information for switch. It is missed in this PR.

File: perl-xCAT/xCAT/data/switchinfo.pm [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# IBM(c) 2007 EPL license http://www.eclipse.org/legal/epl-v10.html
+
+# You can get copycds to recognize new distro DVDs/ISOs (so that you do not have to specify -n and -a)
```
Comment (daniceexi) [inline perl-xCAT/xCAT/data/switchinfo.pm]: did you copy this part of comments from discinfo.pm?

File: xCAT-server/lib/xcat/plugins/nodediscover.pm [hunk 3]
```diff
@@ -232,15 +231,33 @@ sub process_request {
...
+        } elsif($request->{arch}->[0] =~ /armv7l/ and $currboot !~ /onie/) { 
+            $nrtab->setNodeAttribs($node, { netboot => 'onie' });  
         }
     }
 
+    if(defined $request->{nodetype} and $request->{nodetype}->[0] = 'switch' and $request->{_xcat_clientmac}->[0]){
+
+        my $switchestab = xCAT::Table->new('switches');
+        if ($switchestab) {
+            my $switchtype=$xCAT::data::switchinfo::global_mac_identity{substr(lc($request->{_xcat_clientmac}->[0]),0,8)};
```
Comment (daniceexi) [inline xCAT-server/lib/xcat/plugins/nodediscover.pm]: please add comments

File: xCAT/postscripts/documulusdiscovery [hunk 4]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
```
Comment (daniceexi) [inline xCAT/postscripts/documulusdiscovery]: Code is good. But I did not see any comments.

Comment (immarvin) [inline xCAT/postscripts/documulusdiscovery]: added

File: xCAT-server/lib/xcat/plugins/nodediscover.pm (hunk 3 above)
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/nodediscover.pm]: added

File: perl-xCAT/xCAT/data/switchinfo.pm (hunk 2 above)
Comment (immarvin) [inline perl-xCAT/xCAT/data/switchinfo.pm]: corrected

**Comment by zet809** [comment]
Looks good to me, I agree to merge.

**Pull request merged**